src/base/NativeWrapper.sol

```diff
@@ -0,0 +1,34 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.0;
+
+import {IWETH9} from "../interfaces/external/IWETH9.sol";
+import {ActionConstants} from "../libraries/ActionConstants.sol";
+import {ImmutableState} from "./ImmutableState.sol";
+
+/// @title Native Wrapper
+/// @notice Used for wrapping and unwrapping native
+abstract contract NativeWrapper is ImmutableState {
```

Comment:
Why is this inheriting from ImmutableState?

Reply:
OH i see why ig nore me

Reply:
because I need the poolManager address in this contract